src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +156,17 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        if (!$this->getConfig('salt')) {
+            return $identity[$usernameField] . $identity[$passwordField];
+        }
+
+        if (is_string($this->getConfig('salt'))) {
+            $salt = $this->getConfig('salt');
+        } else {
+            $salt = Security::getSalt();
```

Comment:
You could put the before the if to avoid the else.

Reply:
I think that's personal preference. I prefer this:

```
if (...) {
    $x = $a;
} else {
    $x = $b;
}
```

Over this:

```
$x = $b;
if (...) {
    $x = $a;
}
```

(If I understand you correctly.)  
Can we leave this as it is?

Reply:
Ended up changing this anyway because the of conversation below. Resolving this.